packages/amplify_core/lib/src/logger/amplify_logger.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+import 'package:amplify_core/src/logger/level_extension.dart';
+import 'package:logging/logging.dart';
+
+import '../category/amplify_categories.dart';
+import 'log_level.dart';
+
+class AmplifyLogger {
+  static const loggerNamespace = 'AWS.Amplify';
+
+  static final Map<AmplifyLoggerPlugin, StreamSubscription<LogRecord>>
+      _subscriptions = {};
+  final Logger _logger;
+
+  factory AmplifyLogger([String namespace = loggerNamespace]) =>
+      AmplifyLogger._(Logger(namespace));
+
+  factory AmplifyLogger.category(Category category) =>
+      AmplifyLogger._(Logger('$loggerNamespace.${category.name}'));
+
+  AmplifyLogger._(this._logger) {
+    hierarchicalLoggingEnabled = true;
+  }
+
+  void registerPlugin(AmplifyLoggerPlugin plugin) {
```

Comment:
Should these methods be static?

Reply:
No, each logger maintains it's own map of plugins. Due to the hierarchical nature, plugins applied to parent loggers receive all child events so doing `AmplifyLogger().registerPlugin` will receive all Amplify events whereas `AmplifyLogger.category(Category.auth).registerPlugin` registers for only Auth events.

Reply:
Got it. @fjnoyp - API docs would be good here, but could be in a follow up PR.